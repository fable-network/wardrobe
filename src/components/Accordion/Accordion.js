import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import AccordionItem from '../AccordionItem';

const Wrapper = styled.div``;

const Accordion = ({ children }) => <Wrapper>{children}</Wrapper>;

Accordion.Item = AccordionItem;

Accordion.propTypes = {
  /**
   * Array of `Accordion.Item`s.
   */
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
};

export default Accordion;
