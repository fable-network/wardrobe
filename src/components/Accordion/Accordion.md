```jsx
<Accordion>
  <Accordion.Item title="A title"><p>some body text</p></Accordion.Item>
  <Accordion.Item title="Another title"><p>some more body text</p></Accordion.Item>
</Accordion>
```
