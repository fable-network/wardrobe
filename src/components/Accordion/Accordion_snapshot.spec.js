/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import Accordion from './Accordion';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('Accordion Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <Accordion>
          <Accordion.Item title="A title">
            <p>some body text</p>
          </Accordion.Item>
          <Accordion.Item title="Another title">
            <p>some more body text</p>
          </Accordion.Item>
        </Accordion>
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <Accordion>
          <Accordion.Item title="A title">
            <p>some body text</p>
          </Accordion.Item>
          <Accordion.Item title="Another title" defaultOpen={true}>
            <p>some more body text</p>
          </Accordion.Item>
        </Accordion>
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <Accordion>
          <Accordion.Item title="A title" open={true}>
            <p>some body text</p>
          </Accordion.Item>
          <Accordion.Item title="Another title" open={false}>
            <p>some more body text</p>
          </Accordion.Item>
        </Accordion>
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <Accordion>
          <Accordion.Item title="A title" open={true}>
            <p>some body text</p>
          </Accordion.Item>
          <Accordion.Item title="Another title" open={true}>
            <p>some more body text</p>
          </Accordion.Item>
        </Accordion>
      );
      expect(wrapper4).toMatchSnapshot();

    });
  });
});
