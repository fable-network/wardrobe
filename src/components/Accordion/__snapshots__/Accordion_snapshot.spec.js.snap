// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Accordion>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      tabIndex={0}
      title="A title"
    >
      <p>
        some body text
      </p>
    </AccordionItem>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      tabIndex={0}
      title="Another title"
    >
      <p>
        some more body text
      </p>
    </AccordionItem>
  </Accordion>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          tabIndex={0}
          title="A title"
        >
          <p>
            some body text
          </p>
        </AccordionItem>,
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          tabIndex={0}
          title="Another title"
        >
          <p>
            some more body text
          </p>
        </AccordionItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "tabIndex": 0,
          "title": "A title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some body text",
          },
          "ref": null,
          "rendered": "some body text",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some more body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "tabIndex": 0,
          "title": "Another title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some more body text",
          },
          "ref": null,
          "rendered": "some more body text",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            tabIndex={0}
            title="A title"
          >
            <p>
              some body text
            </p>
          </AccordionItem>,
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            tabIndex={0}
            title="Another title"
          >
            <p>
              some more body text
            </p>
          </AccordionItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "tabIndex": 0,
            "title": "A title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some body text",
            },
            "ref": null,
            "rendered": "some body text",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some more body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "tabIndex": 0,
            "title": "Another title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some more body text",
            },
            "ref": null,
            "rendered": "some more body text",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Accordion Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Accordion>
    <AccordionItem
      defaultOpen={true}
      disabled={false}
      onToggle={[Function]}
      tabIndex={0}
      title="A title"
    >
      <p>
        some body text
      </p>
    </AccordionItem>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      tabIndex={0}
      title="Another title"
    >
      <p>
        some more body text
      </p>
    </AccordionItem>
  </Accordion>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AccordionItem
          defaultOpen={true}
          disabled={false}
          onToggle={[Function]}
          tabIndex={0}
          title="A title"
        >
          <p>
            some body text
          </p>
        </AccordionItem>,
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          tabIndex={0}
          title="Another title"
        >
          <p>
            some more body text
          </p>
        </AccordionItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some body text
          </p>,
          "defaultOpen": true,
          "disabled": false,
          "onToggle": [Function],
          "tabIndex": 0,
          "title": "A title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some body text",
          },
          "ref": null,
          "rendered": "some body text",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some more body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "tabIndex": 0,
          "title": "Another title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some more body text",
          },
          "ref": null,
          "rendered": "some more body text",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AccordionItem
            defaultOpen={true}
            disabled={false}
            onToggle={[Function]}
            tabIndex={0}
            title="A title"
          >
            <p>
              some body text
            </p>
          </AccordionItem>,
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            tabIndex={0}
            title="Another title"
          >
            <p>
              some more body text
            </p>
          </AccordionItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some body text
            </p>,
            "defaultOpen": true,
            "disabled": false,
            "onToggle": [Function],
            "tabIndex": 0,
            "title": "A title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some body text",
            },
            "ref": null,
            "rendered": "some body text",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some more body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "tabIndex": 0,
            "title": "Another title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some more body text",
            },
            "ref": null,
            "rendered": "some more body text",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Accordion Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Accordion>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      open={true}
      tabIndex={0}
      title="A title"
    >
      <p>
        some body text
      </p>
    </AccordionItem>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      tabIndex={0}
      title="Another title"
    >
      <p>
        some more body text
      </p>
    </AccordionItem>
  </Accordion>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          open={true}
          tabIndex={0}
          title="A title"
        >
          <p>
            some body text
          </p>
        </AccordionItem>,
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          tabIndex={0}
          title="Another title"
        >
          <p>
            some more body text
          </p>
        </AccordionItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "open": true,
          "tabIndex": 0,
          "title": "A title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some body text",
          },
          "ref": null,
          "rendered": "some body text",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some more body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "tabIndex": 0,
          "title": "Another title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some more body text",
          },
          "ref": null,
          "rendered": "some more body text",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            open={true}
            tabIndex={0}
            title="A title"
          >
            <p>
              some body text
            </p>
          </AccordionItem>,
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            tabIndex={0}
            title="Another title"
          >
            <p>
              some more body text
            </p>
          </AccordionItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "open": true,
            "tabIndex": 0,
            "title": "A title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some body text",
            },
            "ref": null,
            "rendered": "some body text",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some more body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "tabIndex": 0,
            "title": "Another title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some more body text",
            },
            "ref": null,
            "rendered": "some more body text",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Accordion Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Accordion>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      open={true}
      tabIndex={0}
      title="A title"
    >
      <p>
        some body text
      </p>
    </AccordionItem>
    <AccordionItem
      defaultOpen={false}
      disabled={false}
      onToggle={[Function]}
      open={true}
      tabIndex={0}
      title="Another title"
    >
      <p>
        some more body text
      </p>
    </AccordionItem>
  </Accordion>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          open={true}
          tabIndex={0}
          title="A title"
        >
          <p>
            some body text
          </p>
        </AccordionItem>,
        <AccordionItem
          defaultOpen={false}
          disabled={false}
          onToggle={[Function]}
          open={true}
          tabIndex={0}
          title="Another title"
        >
          <p>
            some more body text
          </p>
        </AccordionItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "open": true,
          "tabIndex": 0,
          "title": "A title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some body text",
          },
          "ref": null,
          "rendered": "some body text",
          "type": "p",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            some more body text
          </p>,
          "defaultOpen": false,
          "disabled": false,
          "onToggle": [Function],
          "open": true,
          "tabIndex": 0,
          "title": "Another title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some more body text",
          },
          "ref": null,
          "rendered": "some more body text",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            open={true}
            tabIndex={0}
            title="A title"
          >
            <p>
              some body text
            </p>
          </AccordionItem>,
          <AccordionItem
            defaultOpen={false}
            disabled={false}
            onToggle={[Function]}
            open={true}
            tabIndex={0}
            title="Another title"
          >
            <p>
              some more body text
            </p>
          </AccordionItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "open": true,
            "tabIndex": 0,
            "title": "A title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some body text",
            },
            "ref": null,
            "rendered": "some body text",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              some more body text
            </p>,
            "defaultOpen": false,
            "disabled": false,
            "onToggle": [Function],
            "open": true,
            "tabIndex": 0,
            "title": "Another title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some more body text",
            },
            "ref": null,
            "rendered": "some more body text",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
